webapp/src/Service/ImportProblemService.php

```diff
@@ -926,36 +938,61 @@ public function importProblemFromRequest(Request $request, ?int $contestId = nul
         ];
     }
 
+    /**
+     * @param array{danger: string[], info: string[]} $messages
+     */
+    private function searchAndAddOutputVisualizer(ZipArchive $zip, ?array &$messages, string $externalId, string $visualizerMode, ?Problem $problem): bool
+    {
+        $programStrings = [];
+        $programStrings['package_dir'] = 'output_visualizer/';
+        $programStrings['type'] = 'output visualizer';
+        $programStrings['clash'] = 'visual';
+        return self::helperSearchAndAddProgram($zip, $messages, $externalId, $visualizerMode, $problem, $programStrings);
+    }
+
     /**
      * @param array{danger: string[], info: string[]} $messages
      */
     private function searchAndAddValidator(ZipArchive $zip, ?array &$messages, string $externalId, string $validationMode, ?Problem $problem): bool
     {
-        $validatorFiles = [];
+        $programStrings = [];
+        $programStrings['package_dir'] = 'output_validators/';
+        $programStrings['type'] = 'output validator';
+        $programStrings['clash'] = 'cmp';
```

Comment:
what does `clash` stand for?

Reply:
naming clash, this is copied mostly from how the output_validator is uploaded.